### PR TITLE
Remove type from field to prevent mypy issue

### PR DESCRIPTION
**Pull request by patrick91**

### CODE REVIEW

**Comment by botberry** [comment]
Thanks for adding the `RELEASE.md` file!



Here's a preview of the changelog:

---

This release fixes an issue with mypy when doing the following:

```python
import strawberry

@strawberry.type
class User:
    name: str = strawberry.field(description='Example')
```

File: tests/mypy/test_plugin.yml [hunk 1]
```diff
@@ -36,3 +36,16 @@
...
+- case: test_field
+  main: |
+    import strawberry
+
+    @strawberry.type
+    class User:
+        name: str = strawberry.field(description='Example')
+
+    User(n="Patrick")
+    User(name="Patrick")
```
Comment (BryceBeagle) [inline tests/mypy/test_plugin.yml:49]: Does this line ever get hit?

Comment (patrick91) [reply tests/mypy/test_plugin.yml:48]: yes :) I've inverted the lines so it is more clear that it runs that line too :)

File: strawberry/field.py [hunk 2]
```diff
@@ -70,7 +70,7 @@ def field(
     description: Optional[str] = None,
     permission_classes: Optional[List[Type[BasePermission]]] = None,
     federation: Optional[FederationFieldParams] = None
-) -> Union[Callable, StrawberryField]:
```
Comment (patrick91) [inline strawberry/field.py:73]: @BryceBeagle I wonder if we need this in future 🤔

Comment (BryceBeagle) [reply strawberry/field.py:73]: well hopefully it will just be `-> StrawberryField` in the future

Comment (BryceBeagle) [reply strawberry/field.py:73]: Also, in hindsight, `StrawberryField` _is_ callable, so this is a bit redundant

Comment (BryceBeagle) [reply strawberry/field.py:73]: what I _really_ should have done was
```python
def field(resolver: Optional[Callable[T]], ...) -> Union[Callable[T], StrawberryField]: ...
```

Comment (patrick91) [reply strawberry/field.py:73]: I think that would still break mypy, if we don't update the plugin 😊

for reference `dataclasses.field` is not typed :)

Comment (BryceBeagle) [reply strawberry/field.py:73]: Yeah for sure. I think the long-term solution would be to update the mypy plugin _and_ annotate here. Then we can have our 🎂 and eat it too.

**Pull request merged**